src/components/layout/header.jsx

```diff
@@ -0,0 +1,83 @@
+import styled from "styled-components";
+
+/**
+ * Header Component used with the Layout Component
+ * @author [K. Ehrenclou](https://github.com/kehrenclou)
+ * @param children pass components and namespaced components to header via children
+ * @param mb used to set the header container margin bottom dynamically - default is 32px
+ */
+
+const StyledContainer = styled.div.attrs((props) => ({
+  mb: props.mb || "32px",
+}))`
+  display: grid;
+  grid-template-columns: 1fr minmax(40px, auto) 1fr;
+  grid-template-areas: "left center right";
+  margin-bottom: ${(props) => props.mb};
+`;
+
+const StyledLeft = styled.div`
+  grid-area: left;
+  justify-self: start;
+`;
+const StyledCenter = styled.div`
+  grid-area: center;
+  justify-self: center;
+  justify-content: center;
+  text-align: center;
+`;
+const StyledRight = styled.div`
+  grid-area: right;
+  justify-self: end;
+  column-gap: 12px;
+`;
+
+const StyledTitle = styled.h2`
+  font-weight: 700;
+`;
+const StyledSubtitle = styled.h2`
+  font-weight: 400;
+`;
+
+const left = ({ children }) => {
+  return <StyledLeft>{children}</StyledLeft>;
+};
+const right = ({ children }) => {
+  return (
+    <StyledRight className="d-flex align-items-start">{children}</StyledRight>
+  );
+};
+const center = ({ children }) => {
+  return (
+    <StyledCenter className="d-flex flex-column px-16px gap-12px align-items-center">
+      {children}
+    </StyledCenter>
+  );
+};
+const title = ({ title }) => {
+  return (
+    <StyledTitle className="fs-24px m-0 align-self-center">{title}</StyledTitle>
+  );
+};
+const subtitle = ({ subtitle }) => {
+  return (
+    <StyledSubtitle className="fs-12px m-0 text-white text-opacity-70">
+      {subtitle}
+    </StyledSubtitle>
+  );
+};
+const Header = ({ children, mb }) => {
+  return (
+    <StyledContainer mb={mb} className="text-white text-opacity-90">
```

Comment:
having trouble with this.  I've been able to get the classname to get paased to the correct component, but it is not rendering the new value, it is still keeping the default 32px but not rendering the new margin bottom value.

![image](https://user-images.githubusercontent.com/98502327/219877933-ebf6d706-f556-4e24-8d0e-9c3b46c2b15f.png)

Reply:
`export const Header = ({ children, className }) => {
  return (
    <StyledContainer
      className={cx(`text-white text-opacity-90 mb-32px`, className)}
    >
      {children}
    </StyledContainer>
  );
};`

Reply:
```
//header.jsx
export const Header = ({ children, className }) => {
  return (
    <StyledContainer
      className={cx(`text-white text-opacity-90 mb-32px`, className)}
    >
      {children}
    </StyledContainer>
  );
};
```

Reply:
//layout.jsx
```
export const Layout = ({ header, body, footer, className }) => {
  return (
    <Container>
      {header && <Header className={className}>{header}</Header>}
      <Body>{body}</Body>
      {footer && <Footer>{footer}</Footer>}
    </Container>
  );
};
```

Reply:
example of it being used that resulted in screen shot above
//user-infor.jsx
```
export function UserInfo({ onBackClick, onForwardClick }) {
  return (
    <Layout
      className="mb-0"
      header={
        <>
          <Header.Left>
            <Button onClick={onBackClick}>**</Button>
          </Header.Left>
          <Header.Center>
            <Header.Title title="Sign up" />
            <Header.SubTitle subtitle="Step 1/4" />
          </Header.Center>{" "}
        </>
      }
      body="insert form component here"
      footer={<Button onClick={onForwardClick}>Continue</Button>} //replace buton with step button
    ></Layout>
  );
}
```

Reply:
I see-- including a default might be tricky now that I think about it. You may have to not use a default and just always pass in a classname for margins; I think this is fine though since a margin is sort of "outside" of the component so it shouldn't really have a pre-baked value IMO

Reply:
done